stock_picking_package_info/models/stock_quant_package.py

```diff
@@ -0,0 +1,198 @@
+    )
+    total_est_weight = fields.Float(
+        string='Total Estimated Weight',
+        store=True,
+        compute='_compute_total_est_weights',
+        digits=dp.get_precision('Stock Weight'),
+    )
+    total_est_weight_net = fields.Float(
+        string='Total Estimated Net Weight',
+        store=True,
+        compute='_compute_total_est_weight_net',
+        digits=dp.get_precision('Stock Weight'),
+    )
+    real_weight = fields.Float(
+        store=True,
+        compute='_compute_total_est_weights',
+        digits=dp.get_precision('Stock Weight'),
+    )
+
+    @api.multi
+    @api.depends(
+        'quant_ids',
+        'quant_ids.total_weight',
+        'quant_ids.total_weight_net',
+    )
+    def _compute_total_weights(self):
+        for rec_id in self:
+            rec_id.total_weight, rec_id.total_weight_net = 0, 0
+
+            for quant_id in rec_id.quant_ids:
+                rec_id.total_weight += quant_id.total_weight
+                rec_id.total_weight_net += quant_id.total_weight_net
+
+    @api.multi
+    @api.depends(
+        'total_weight',
+        'empty_weight',
+        'children_ids',
+        'children_ids.total_weight',
+        'children_ids.empty_weight',
+    )
+    def _compute_total_est_weights(self):
+        for rec_id in self:
+            estim_sum = (
+                rec_id.total_weight + rec_id.empty_weight
+            )
+            for child_id in rec_id.children_ids:
+                estim_sum += (
+                    child_id.total_weight + child_id.empty_weight
+                )
+            rec_id.total_est_weight = estim_sum
+            rec_id.real_weight = estim_sum
+
+    @api.multi
+    @api.depends(
+        'total_weight_net',
+        'empty_weight',
+        'children_ids',
+        'children_ids.total_weight_net',
+        'children_ids.empty_weight',
+    )
+    def _compute_total_est_weight_net(self):
+        for rec_id in self:
+            estim_sum = (
+                rec_id.total_weight_net + rec_id.empty_weight
+            )
+            for child_id in rec_id.children_ids:
+                estim_sum += (
+                    child_id.total_weight_net + child_id.empty_weight
+                )
+            rec_id.total_est_weight_net = estim_sum
+
+    @api.multi
+    @api.depends('height', 'width', 'length')
+    def _compute_permitted_volume(self):
+        for rec_id in self:
+            rec_id.permitted_volume = (
+                rec_id.height * rec_id.width * rec_id.length
+            )
+
+    @api.multi
+    @api.depends('quant_ids', 'quant_ids.total_volume')
+    def _compute_total_volume(self):
+        for rec_id in self:
+            rec_id.total_volume = 0
+            for quant_id in rec_id.quant_ids:
+                rec_id.total_volume += quant_id.total_volume
+                _logger.debug('-' * 20)
+                _logger.debug('Total Volume')
+                _logger.debug(rec_id.total_volume)
```

Comment:
I think those may have been added for testing purposes. Can probably just be removed.